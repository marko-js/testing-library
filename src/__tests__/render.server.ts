import { render, screen, fireEvent, cleanup } from "..";
import Counter from "./fixtures/counter.marko";
import LegacyCounter from "./fixtures/legacy-counter";
import Clickable from "./fixtures/clickable.marko";
import HelloName from "./fixtures/hello-name.marko";

test("renders static content in a document with a browser context", async () => {
  const { getByText, container } = await render(Counter);
  expect(
    expect(getByText("Value: 0")).toHaveProperty([
      "ownerDocument",
      "defaultView",
    ])
  ).not.toBeNull();

  expect(container.firstElementChild).toHaveAttribute("class", "counter");
});

test("renders static content from a Marko 3 component", async () => {
  const { getByText } = await render(LegacyCounter);
  expect(
    expect(getByText("Value: 0")).toHaveProperty([
      "ownerDocument",
      "defaultView",
    ])
  ).not.toBeNull();
});

test("global cleanup removes content from the document", async () => {
  await render(Counter);
  expect(screen.queryAllByText(/Value: 0/)).toHaveLength(1);
  cleanup();
  expect(screen.queryAllByText(/Value: 0/)).toHaveLength(0);
});

test("local cleanup removes single component from the document", async () => {
  const { cleanup } = await render(Counter);
  expect(screen.queryAllByText(/Value: 0/)).toHaveLength(1);
  cleanup();
  expect(screen.queryAllByText(/Value: 0/)).toHaveLength(0);
  expect(() => cleanup()).toThrowErrorMatchingSnapshot();
});

test("fails when rerendering", async () => {
  const { rerender } = await render(HelloName, { name: "Michael" });
  await expect(
    rerender({ name: "Dylan" })
  ).rejects.toThrowErrorMatchingInlineSnapshot(
    `"Components cannot re-render on the server side"`
  );
});

test("fails when checking emitted events", async () => {
  const { emitted } = await render(Clickable);
  expect(() => emitted("button-click")).toThrowErrorMatchingInlineSnapshot(
    `"Components should not emit events on the server side"`
  );
});

test("fails when emitting events", async () => {
  const { getByText } = await render(Counter);
  await expect(
    fireEvent.click(getByText("Increment"))
  ).rejects.toThrowErrorMatchingInlineSnapshot(
    `"Cannot fire events when testing on the server side. Please use @marko/testing-library in a browser environment."`
  );
});
