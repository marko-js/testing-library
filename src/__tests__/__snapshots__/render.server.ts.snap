// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`local cleanup removes single component from the document 1`] = `"Component was already destroyed before cleanup called."`;
